velodyne_pointcloud/include/velodyne_pointcloud/datacontainerbase.h

```diff
@@ -181,13 +174,39 @@ class DataContainerBase
     Eigen::Vector3f eigen_origin;
     vectorTfToEigen(transform.getOrigin(), eigen_origin);
     Eigen::Translation3f translation(eigen_origin);
-    transformation = translation * rotation;
+    htm = translation * rotation;
     return true;
   }
 
+  inline bool computeTransformToTarget(const ros::Time &scan_time)
+  {
+    std::string &source_frame = config_.fixed_frame.empty()
+                                    ? cloud.header.frame_id
+                                    : config_.fixed_frame;
+    return !(!config_.target_frame.empty() &&
+             !lookupHTM(htm_to_target, config_.target_frame, source_frame,
+                        scan_time));
+  }
+
+  inline bool computeTransformToFixed(const ros::Time &packet_time)
+  {
+    std::string &source_frame = cloud.header.frame_id;
+    return !(!config_.fixed_frame.empty() &&
+             !lookupHTM(htm_to_fixed, config_.fixed_frame, source_frame,
+                        packet_time));
+  }
+
   inline void transformPoint(float& x, float& y, float& z)
   {
-    Eigen::Vector3f p = transformation * Eigen::Vector3f(x, y, z);
+    Eigen::Vector3f p = Eigen::Vector3f(x, y, z);
+    if (!config_.fixed_frame.empty())
+    {
+      p = htm_to_fixed * p;
+    }
+    if (!config_.target_frame.empty())
+    {
+      p = htm_to_target * p;
+    }
```

Comment:
Only perform a transform when fixed_frame is specified. It should be just specified if one wants to use ego motion compensation. Otherwise it should be left empty in the params. Note that `htm_to_fixed` was only calculated when fixed_frame is not empty (see `computeTransformToFixed`)

Reply:
Same holds for target frame